test/test_messaging.py

```diff
@@ -247,6 +247,41 @@ def test_publish_custom_headers(
     ]
 
 
+@pytest.mark.usefixtures("predictable_call_ids")
+def test_publish_custom_headers_added_in_current_request(
+    mock_container, mock_producer, rabbit_config
+):
+    container = mock_container
+    container.config = rabbit_config
+    container.service_name = "srcservice"
+
+    ctx_data = {'language': 'en', 'customheader1': 'customvalue1'}
+    service = Mock()
+    worker_ctx = WorkerContext(
+        container, service, DummyProvider('method'), data=ctx_data
+    )
+
+    publisher = Publisher(queue=foobar_queue).bind(container, "publish")
+    publisher.setup()
+
+    # test publish
+    msg = "msg"
+    headers = {'nameko.language': 'en',
```

Comment:
Can we call this `expected_headers` and define it _after_ the call to `publish`?